### ISSUE TITLE
Reverse lookup, City doubled as County in JSON/XML response

### ISSUE THREAD

**Issue by InI4** [openstreetmap/Nominatim#899]
I've seen this elsewhere, but stick to Stockholm as example:

`https://nominatim.openstreetmap.org/reverse?format=json&lat=59.331088&lon=18.093109&zoom=9&accept-language=en&addressdetails=1
`
gives:
```
{
    "place_id": "180047801",
    "licence": "Data © OpenStreetMap contributors, ODbL 1.0. http://www.openstreetmap.org/copyright",
    "osm_type": "relation",
    "osm_id": "6236659",
    "lat": "59.3244528",
    "lon": "18.1152567352819",
    "display_name": "Djurgårdsstaden, Djurgården, Östermalms stadsdelsområde, Stockholm, Landskapet Uppland, Stockholms län, Svealand, 115 21, Sweden",
    "address": {
        "neighbourhood": "Djurgårdsstaden",
        "suburb": "Djurgården",
        "city_district": "Östermalms stadsdelsområde",
        "city": "Stockholm",
        "county": "Stockholm",
        "state_district": "Landskapet Uppland",
        "state": "Stockholms län",
        "postcode": "115 21",
        "country": "Sweden",
        "country_code": "se"
    },
    "boundingbox": [
        "59.3176312",
        "59.3311452",
        "18.0831362",
        "18.1382765"
    ]
}
```

Note the difference between the `display_name` field and the `address structure`: The latter contains Stockholm twice, both as `city` and as `county`. The true hierarchy is hard to be obtained automatically,

Other requests, for example

`https://nominatim.openstreetmap.org/reverse?format=json&lat=59.294643&lon=17.928314&zoom=42&accept-language=en&addressdetails=1
`
contain Stockholm only as county, but not as city.
```
{
    "place_id": "166199146",
    "licence": "Data © OpenStreetMap contributors, ODbL 1.0. http://www.openstreetmap.org/copyright",
    "osm_type": "way",
    "osm_id": "451482944",
    "lat": "59.29462745",
    "lon": "17.9288074",
    "display_name": "Pizza No.1, Stora Sällskapets Väg, Bredäng, Skärholmens stadsdelsområde, Stockholm, Landskapet Södermanland, Stockholms län, Svealand, 127 61, Sweden",
    "address": {
        "fast_food": "Pizza No.1",
        "road": "Stora Sällskapets Väg",
        "suburb": "Bredäng",
        "city_district": "Skärholmens stadsdelsområde",
        "county": "Stockholm",
        "state_district": "Landskapet Södermanland",
        "state": "Stockholms län",
        "postcode": "127 61",
        "country": "Sweden",
        "country_code": "se"
    },
    "boundingbox": [
        "59.2945908",
        "59.2946641",
        "17.928735",
        "17.9288798"
    ]
}
```
This is not correct, but suffices to obtain a valid nesting of structures.

XML files show the same problem.

**Comment by lonvia** [comment]
The fields in address are ordered. That is admittedly not very helpful in the json output because JSON hashes have no order but it works without issues in XML. JSON will be fixed if we ever get to release version 2 of the API. Until then you can either use a JSON library that preserves order or could keep a list of which address part belongs to which level, see https://wiki.openstreetmap.org/wiki/Nominatim#Hierarchy.

**Comment by InI4** [comment]
I am not that much concerned about the actual order of fields, my scripts can handle that part, but about the doubling of inclusions. 

In the upper sample there is
```
        "city": "Stockholm",
        "county": "Stockholm",

```
but there is no such thing as a county Stockholm, only a city. I'd need to dig for the examples in my logs, but it seems like locations which belong to the city of Stockholm get classified in 3 different ways:

1. City of Stockholm (sample missing)
2. County of Stockholm (sample 2 above)
3. County and City of Stockholm (sample 1 above)

Actually my scripts only go for nesting, so while irritated about case 1. and 2., the last case gets me in trouble, as it introduces an extra layer. For case 3, the nesting of Stockholm does not seem to happen in the `display_name` field, which makes me hope, it could be some kind of a bug in the JSON?!

**Comment by lonvia** [comment]
This happens when the local use of admin levels in boundary relations doesn't match Nominatim's (globally used) assumptions how they match to place types. This can only be fixed if #213 ever gets implemented.

In this particular case you can help Nominatim along to find the right matching by changing https://www.openstreetmap.org/relation/398021 and setting the member role of the Stockholm node to 'label' instead of 'admin_centre". but do that only if that relation really represents the administrative boundary of the city of Stockholm.
